src/main/java/com/commercetools/sync/inventories/InventorySync.java

```diff
@@ -343,20 +383,23 @@ public InventorySyncStatistics getStatistics() {
     }
 
     /**
-     * Returns {@link CompletionStage} instance which may contain mapping of {@link ChannelKeyExtractor} to
-     * {@link InventoryEntry} of instances existing in a CTP project. Instances are fetched from API by skus, that
-     * corresponds to skus present in {@code drafts}. If fetching existing instances results in exception then
-     * returned {@link CompletionStage} contains such exception.
+     * Tries to fetch from a CTP a list of inventory entries of SKUs present in {@code drafts}.
+     * If operation succeed then {@link CompletionStage} containing {@link Optional} with fetched {@link List} is
+     * returned, otherwise error callback function is executed and {@link CompletionStage} with empty {@link Optional}
+     * is returned.
      *
      * @param drafts {@link List} of drafts
-     * @return {@link CompletionStage} instance which contains either {@link Map} of {@link ChannelKeyExtractor} to
-     *      {@link InventoryEntry} of instances existing in a CTP project that correspond to passed {@code drafts} by
-     *      sku comparision, or exception occurred during fetching existing inventory entries
+     * @return {@link CompletionStage} instance with {@link Optional} that may contain list of inventory entries
      */
-    private CompletionStage<List<InventoryEntry>> fetchExistingInventories(@Nonnull final List<InventoryEntryDraft>
-                                                                               drafts) {
+    private CompletionStage<Optional<List<InventoryEntry>>> fetchExistingInventories(@Nonnull final
+                                                                                     List<InventoryEntryDraft> drafts) {
         final Set<String> skus = extractSkus(drafts);
-        return inventoryService.fetchInventoryEntriesBySkus(skus);
+        return inventoryService.fetchInventoryEntriesBySkus(skus)
+            .thenApply(Optional::of)
```

Comment:
@heshamMassoud An `exceptionally` block is already present inside `fetchExistingInventories` - right after invocation of service's method. Exception handling should be close to the source of possible errors (service calls) so I wouldn't move the `exceptionally` block outside this function. Since the error is already handled, then method is expected to return something. The `Optional` fits into my needs here. The alternative is to return a nullable resul, but I think `Optional` is better than that.